frontend/src/app/testQueue/addToQueue/AddToQueueSearch.jsx

```diff
@@ -89,60 +95,32 @@ const AddToQueueSearchBox = ({ refetchQueue, facilityId, patientsInQueue }) => {
     appInsights,
     "Add Patient to Queue"
   );
-
-  const { data, loading, error } = useQuery(QUERY_PATIENT, {
+  const [queryString, setQueryString] = useState("");
+  const { data, error } = useQuery(QUERY_PATIENT, {
     fetchPolicy: "no-cache",
-    variables: { facilityId },
+    variables: { facilityId, searchTerm: queryString },
   });
-
   const [mutationError, updateMutationError] = useState(null);
   const [addPatientToQueue] = useMutation(ADD_PATIENT_TO_QUEUE);
   const [updateAoe] = useMutation(UPDATE_AOE);
-  const [queryString, setQueryString] = useState("");
-  const [suggestions, updateSuggestions] = useState([]);
 
-  if (loading) {
-    return <p> Loading patient data... </p>;
-  }
   if (error) {
     throw error;
   }
   if (mutationError) {
     throw mutationError;
   }
 
-  let shouldShowSuggestions = queryString.length >= MIN_SEARCH_CHARACTER_COUNT;
-
-  const getSuggestionsFromQueryString = (queryString) => {
-    if (data && data.patients) {
-      let formattedQueryString = queryString.toLowerCase();
-      let searchResults = data.patients.filter((patient) => {
-        return (
-          displayFullName(
-            patient.firstName,
-            patient.middleName,
-            patient.lastName
-          )
-            .toLowerCase()
-            .indexOf(formattedQueryString) > -1
-        );
-      });
-      return searchResults;
-    }
-    return [];
-  };
-
   const onInputChange = (event) => {
     let newValue = event.target.value;
+    // debouncing helps reduce traffic, but introduces a lag in results.
```

Comment:
I would still vote for a small debounce, maybe a couple hundred milliseconds. Seems like a _lot_ of unnecessary network traffic if the search fires on every single keystroke. A really small debounce would probably be imperceptible to users but really cut down on requests.

I would also implement the min character length when debouncing the request.

Reply:
Thanks for adding debounce. 500ms may seem a bit laggy, but we can easily change it in the future.